### PR TITLE
Fix create rollout + group + update

### PR DESCRIPTION
**Pull request by duanxianbo**
[(task:19552)](https://staging.teamyapp.com/teams/107/tasks/19552)

### CODE REVIEW

File: src/storage/syncer/stateMutators.ts [hunk 1]
```diff
@@ -1,3 +1,5 @@
+import { merge } from 'lodash';
+
```
Comment (duanxianbo) [inline src/storage/syncer/stateMutators.ts]:
```suggestion
```

File: src/components/pages/Marketplace/Modals/CreateRolloutView.component.tsx [hunk 2]
```diff
@@ -398,6 +398,12 @@ export class CreateRolloutViewComponent extends Component<Props, State> {
     };
 
     onCreateRolloutClick = async () => {
+        if (!this.state.staticVersionSelectorInput?.versionNumber 
+        && !this.state.experimentVersionSelectorInput?.versionNumbers?.length) {
```
Comment (naturetime10) [inline src/components/pages/Marketplace/Modals/CreateRolloutView.component.tsx]: Can we separate null check for `this.state.experimentVersionSelectorInput?.versionNumbers` and checking it's `length`?

File: src/components/pages/Marketplace/Modals/EditRolloutView.component.tsx [hunk 3]
```diff
@@ -196,6 +196,11 @@ export function EditRolloutViewComponent() {
                 };
 
             const handleSaveClicked = async () => {
+                if (!staticVersionSelectorInput?.versionNumber && !experimentVersionSelectorInput?.versionNumbers.length) {
```
Comment (naturetime10) [inline src/components/pages/Marketplace/Modals/EditRolloutView.component.tsx]: Can we separate null check for `this.state.experimentVersionSelectorInput?.versionNumbers` and checking it's `length`?

File: src/components/pages/Marketplace/Modals/ManageAppModal/RolloutsTab.component.tsx [hunk 4]
```diff
@@ -82,14 +82,14 @@ export function RolloutsTabComponent(props: Props) {
         activatorId: number,
         input: UpdateActivatorInput,
     ) => {
```
Comment (naturetime10) [inline src/components/pages/Marketplace/Modals/ManageAppModal/RolloutsTab.component.tsx]: Can include return type here?

File: src/components/pages/Marketplace/Modals/ManageAppModal/RolloutsTab.component.tsx [hunk 5]
```diff
@@ -82,14 +82,14 @@ export function RolloutsTabComponent(props: Props) {
...
-        props.deps.stateSyncer.updateActivator(activatorId, input);
+        return props.deps.stateSyncer.updateActivator(activatorId, input);
     };
 
     const saveVersionSelector = async (
         versionSelectorId: number,
         input: UpdateVersionSelectorInput,
     ) => {
```
Comment (naturetime10) [inline src/components/pages/Marketplace/Modals/ManageAppModal/RolloutsTab.component.tsx]: Can include return type here?

**Review by naturetime10 — CHANGES_REQUESTED** [review]

File: src/storage/syncer/stateMutators.ts [hunk 6]
```diff
@@ -908,29 +909,43 @@ export function mergeGroup(
...
+    remoteRolloutId: number,
+    orderIndex: number,
+) {
+    if (
+        !currAppState.groupRolloutRelations.some((groupRolloutRelation) => {
+            return (
+                groupRolloutRelation.groupId === remoteGroupId &&
+                groupRolloutRelation.rolloutId === remoteRolloutId
+            );
+        })
```
Comment (naturetime10) [inline src/storage/syncer/stateMutators.ts]: Can we extract this into variable?

**Review by naturetime10 — CHANGES_REQUESTED** [review]

**Review by naturetime10 — APPROVED** [review]

**Pull request merged**